modules/acquia_cms_headless/src/Service/StarterkitNextjsService.php

```diff
@@ -298,7 +298,9 @@ public function createHeadlessUser() {
         $user->set("init", $email);
         $user->set("preferred_langcode", $language);
         $user->activate();
-        $user->addRole('headless');
+        foreach ($this->getAllSiteRoles() as $role) {
```

Comment:
I think this could benefit from a comment that explains why we are granting all of the roles for this user.